config/orte_setup_debugger_flags.m4

```diff
@@ -53,4 +68,31 @@ AC_DEFUN([ORTE_SETUP_DEBUGGER_FLAGS],[
 
     AC_SUBST(CFLAGS_WITHOUT_OPTFLAGS)
     AC_SUBST(DEBUGGER_CFLAGS)
+
+    # Check for compiler specific flag to add in unwind information.
+    # This is needed when attaching using MPIR to unwind back to the
+    # user's main function. Certain optimisations can prevent GDB from
+    # producing a stack when explicit unwind information is unavailable.
+    # This is implied by -g, but we want to save space and don't need
+    # full debug symbols.
+    foundFlags=no
+    MPIR_UNWIND_CFLAGS=
+
+    # Test for GNU and Intel compiler flags
+    _ORTE_SETUP_DEBUGGER_FLAGS_TRY_CFLAGS([-fasynchronous-unwind-tables],
+       [MPIR_UNWIND_CFLAGS=-fasynchronous-unwind-tables foundFlags=yes])
+
+    # Test for PGI compiler flags
+    if test $foundFlags = no; then
+        _ORTE_SETUP_DEBUGGER_FLAGS_TRY_CFLAGS([-Meh_frame -Mframe],
+            [MPIR_UNWIND_CFLAGS="-Meh_frame -Mframe" foundFlags=yes])
+    fi
+
+    # Finally fall back to -g if neither of the above were available
+    if test $foundFlags = no; then
+        _ORTE_SETUP_DEBUGGER_FLAGS_TRY_CFLAGS([-g],
+            [MPIR_UNWIND_CFLAGS=-g foundFlags=yes])
+    fi
```

Comment:
Minor nit: could you OPAL_VAR_SCOPE_PUSH / OPAL_VAR_SCOPE_POP for `foundFlags`?

Those macros do a few things:

1. Make sure `foundFlags` is not already set at the beginning (i.e., make sure we're not causing a side effect now or later)
1. Unset `foundFlags` at the end

Reply:
Fixed